blog/2024-05-02-react-sdk-release.md

```diff
@@ -0,0 +1,118 @@
+
+<!--truncate-->
+
+## Features
+
+### Context Provider
+
+The OpenFeature client is a lightweight abstraction used to evaluate feature flags, implemented by every SDK.
+Essentially, they represent a _scope_ for feature flag evaluations.
+React's [_context provider_](https://react.dev/reference/react/createContext#provider) concept provides a means of sharing data and objects across multiple components without explicitly passing them through [_props_](https://react.dev/learn/passing-props-to-a-component).
+A context provider is an ideal means of providing a bounded scope for flag evaluations by associating a react context with an OpenFeature client; the OpenFeature React SDKs `OpenFeatureProvider` does exactly this.
+Within the scope of this React provider, all flag evaluation hooks will make use of the bound OpenFeature client and its associated [OpenFeature provider](https://openfeature.dev/specification/glossary#provider).
+
+```tsx
+function App() {
+  return (
+    <OpenFeatureProvider>
+      {/* components in this scope share the same OpenFeature client and OpenFeature Provider */}
+      <MyPage/>
+    </OpenFeatureProvider>
+  );
+}
+```
+
+### Re-rendering with Flag Changes
+
+The most fundament feature of the React SDK is support for re-rending components using feature flags when the effective values of those flags change.
+Flag value changes could be the result of changes in the evaluation context which is the basis for flag evaluations rules, or change in the flag definitions themselves.
+Changes in the evaluation context are best exemplified by changes to user properties in a application; for example, a user logging in, adding something to their cart, or changing their subscriber level.
+Any of these examples could mean that we need to re-evaluate feature flags in light of this new information.
+The flag evaluation hooks provided by the React SDK do this automatically (by default) when the bound provider emits a [_PROVIDER_CONTEXT_CHANGED_](https://openfeature.dev/specification/types#provider-events) event.
+
+```tsx
+function MyPage() {
+  // the "value" returned here is always up-to-date with changes in rules or contextual data
+  const { value: showNewMessage } = useFlag("new-message", false);
+  return (
+    <div>
+      <header>
+        {showNewMessage ? (
+          <p>Welcome to this OpenFeature-enabled React app!</p>
+        ) : (
+          <p>Welcome to this React app.</p>
+        )}
+      </header>
+    </div>
+  );
+}
+```
+Additionally, some feature flag solutions support realtime updates when flag configurations themselves are changed and the bound provider emits a [PROVIDER_CONFIGURATION_CHANGED](https://openfeature.dev/specification/types#provider-events) event.
+The React SDK also supports re-rendering in these cases, so that a change in a flag definition results in those changes immediately being reflected in the user interface.
```

Comment:
```suggestion
The React SDK also supports re-rendering in these cases so that a change in a flag definition immediately reflects those changes in the user interface.
```